src/main/java/com/google/cloud/resourcemanager/OperationInfo.java

```diff
@@ -0,0 +1,276 @@
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      Status status = (Status) o;
+      return Objects.equals(code, status.code)
+          && Objects.equals(details, status.details)
+          && Objects.equals(message, status.message);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(code, details, message);
+    }
+
+    com.google.api.services.cloudresourcemanager.model.Status toProtobuf() {
+      return new com.google.api.services.cloudresourcemanager.model.Status()
+          .setCode(code)
+          .setMessage(message)
+          .setDetails(details);
+    }
+
+    static Status fromProtobuf(com.google.api.services.cloudresourcemanager.model.Status status) {
+      return new Status(status.getCode(), status.getMessage(), status.getDetails());
+    }
+  }
+
+  static class Builder {
+
+    private Boolean done;
+    private Status error;
+    private Map<String, Object> metadata;
+    private String name;
+    private Map<String, Object> response;
+
+    Builder(String name) {
+      this.name = name;
+    }
+
+    Builder(OperationInfo info) {
+      name = info.name;
+      done = info.done;
+      metadata = info.metadata;
+      response = info.response;
+      error = info.error;
+    }
+
+    Builder setName(String name) {
+      this.name = name;
+      return this;
+    }
+
+    Builder setDone(Boolean done) {
+      this.done = done;
+      return this;
+    }
+
+    Builder setMetadata(Map<String, Object> metadata) {
+      this.metadata = metadata;
+      return this;
+    }
+
+    Builder setResponse(Map<String, Object> response) {
+      this.response = response;
+      return this;
+    }
+
+    Builder setError(Status error) {
+      this.error = error;
+      return this;
+    }
+
+    OperationInfo build() {
+      checkNotNull(name);
+      return new OperationInfo(this);
+    }
+  }
+
+  OperationInfo(Builder builder) {
+    name = builder.name;
+    done = builder.done;
+    metadata = builder.metadata;
+    response = builder.response;
+    error = builder.error;
+  }
+  /**
+   * If the value is false, it means the operation is still in progress. If true, the operation is
+   * completed, and either error or response is available.
+   */
+  public Boolean getDone() {
+    return done;
+  }
+
+  /** Returns the error result of the operation in case of failure or cancellation. */
+  public Status getError() {
+    return error;
+  }
+
+  /**
+   * Returns the service-specific metadata associated with the operation. It typically contains
+   * progress information and common metadata such as create time.
+   */
+  public Map<String, Object> getMetadata() {
+    return metadata;
+  }
+
+  /**
+   * Returns the server-assigned name, which is only unique within the same service that originally
+   * returns it.
+   */
+  public String getName() {
+    return name;
+  }
+
+  /** Returns the normal response of the operation in case of success. */
```

Comment:
What does it return if the call failed?

Reply:
Returns the error result of the operation in case of failure or cancellation.